cmd/agent/app/reporter/grpc/builder.go

```diff
@@ -50,6 +50,8 @@ type ConnBuilder struct {
 	Discoverer        discovery.Discoverer
 
 	AdditionalDialOptions []grpc.DialOption
+
+	Context context.Context
```

Comment:
ConnBuilder is a configuration object, it has no business holding cancellation context. The context should be owned and pass in by whomever uses the builder to construct connections.

Reply:
Got it